core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2132,6 +2152,26 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  //future-y version
+  private def requireAccessF[T](workspace: Workspace, requiredAction: SamResourceAction)(codeBlock: => Future[T]): Future[T] = {
```

Comment:
We now have two versions of requireAccess that both have extremely similar code. It would be nice if, in the future, we could get rid of calling Sam from inside db transactions - but that's definitely outside the scope of this ticket.

Reply:
yeah -- it was a choice of "don't add another sam call inside a txn" or "write a duplicated version of this function".